sdk/src/commitment_config.rs

```diff
@@ -46,11 +46,28 @@ impl CommitmentConfig {
 
 #[derive(Serialize, Deserialize, Clone, Copy, Debug, PartialEq, Eq, Hash)]
 #[serde(rename_all = "camelCase")]
+/// An attribute of a slot. It describes how finalized a block is at some point in time. For example, a slot
+/// is said to be at the max level immediately after the cluster recognizes the block at that slot as
+/// finalized. When querying the ledger state, use lower levels of commitment to report progress and higher
+/// levels to ensure state changes will not be rolled back.
 pub enum CommitmentLevel {
+    /// The highest slot confirmed by the cluster as having reached `MAX_LOCKOUT_HISTORY` confirmations. The
+    /// validator will not recognize forks below this slot.
     Max,
+
+    /// The highest slot of the heaviest fork. Ledger state at this slot is not derived from a finalized
+    /// block, but if multiple forks are present, is from the fork the validator believes is most likely
+    /// to finalize.
     Recent,
+
+    /// The highest slot of a block having reached `MAX_LOCKOUT_HISTORY` confirmations.
```

Comment:
I think we should talk about votes instead of confirmations here to more clearly differentiate from `Max`. Something like: "The highest slot that has reached max vote lockout by the connected node"